shared/protocol/src/main/java/io/pravega/shared/metrics/ClientMetricTags.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.metrics;
+
+import com.google.common.base.Preconditions;
+
+public class ClientMetricTags {
+
+    // Metric Tag Names
+    private static final String TAG_SCOPE = "scope";
+    private static final String TAG_STREAM = "stream";
+    private static final String TAG_SEGMENT = "segment";
+    private static final String TAG_TRANSACTION = "transaction";
+    private static final String TAG_EPOCH = "epoch";
+
+    private static final String TRANSACTION_DELIMITER = "#transaction.";
+    private static final String EPOCH_DELIMITER = ".#epoch.";
+
+    /**
+     * Generate segment tags (string array) on the input fully qualified segment name to be associated with a metric.
+     *
+     * @param qualifiedSegmentName fully qualified segment name.
+     * @return string array as segment tag of metric.
+     */
+    public static String[] segmentTags(String qualifiedSegmentName) {
+        Preconditions.checkNotNull(qualifiedSegmentName);
+        String[] tags = {TAG_SCOPE, null, TAG_STREAM, null, TAG_SEGMENT, null, TAG_EPOCH, null};
+
+        String segmentBaseName = getSegmentBaseName(qualifiedSegmentName);
+        String[] tokens = segmentBaseName.split("[/]");
```

Comment:
This should probably rely on the same split code as in Stream or Segment so that it's guaranteed to be consistent. We also created NameUtils and SegmentNameUtils. Maybe something could be refactored into one of those.

Reply:
done.